connect-inject/handler.go

```diff
@@ -504,6 +548,173 @@ func (h *Handler) defaultAnnotations(pod *corev1.Pod, patches *[]jsonpatch.JsonP
 	return nil
 }
 
+// enableMetrics returns the default value in the handler, or overrides that
+// with the annotation if provided.
+func (h *Handler) enableMetrics(pod *corev1.Pod) (bool, error) {
+	enabled := h.DefaultEnableMetrics
+	if raw, ok := pod.Annotations[annotationEnableMetrics]; ok && raw != "" {
+		enableMetrics, err := strconv.ParseBool(raw)
+		if err != nil {
+			return false, fmt.Errorf("%s annotation value of %s was invalid: %s", annotationEnableMetrics, raw, err)
+		}
+		enabled = enableMetrics
+	}
+	return enabled, nil
+}
+
+// enableMetricsMerging returns the default value in the handler, or overrides
+// that with the annotation if provided.
+func (h *Handler) enableMetricsMerging(pod *corev1.Pod) (bool, error) {
+	enabled := h.DefaultEnableMetricsMerging
+	if raw, ok := pod.Annotations[annotationEnableMetricsMerging]; ok && raw != "" {
+		enableMetricsMerging, err := strconv.ParseBool(raw)
+		if err != nil {
+			return false, fmt.Errorf("%s annotation value of %s was invalid: %s", annotationEnableMetricsMerging, raw, err)
+		}
+		enabled = enableMetricsMerging
+	}
+	return enabled, nil
+}
+
+// mergedMetricsPort returns the default value in the handler, or overrides
+// that with the annotation if provided.
+func (h *Handler) mergedMetricsPort(pod *corev1.Pod) (string, error) {
+	defaultMergedMetricsPort := h.DefaultMergedMetricsPort
+
+	return determineAndValidatePort(pod, annotationMergedMetricsPort, defaultMergedMetricsPort)
+}
+
+// prometheusScrapePort returns the default value in the handler, or overrides
+// that with the annotation if provided.
+func (h *Handler) prometheusScrapePort(pod *corev1.Pod) (string, error) {
+	defaultPrometheusScrapePort := h.DefaultPrometheusScrapePort
+
+	return determineAndValidatePort(pod, annotationPrometheusScrapePort, defaultPrometheusScrapePort)
+}
+
+// prometheusScrapePath returns the default value in the handler, or overrides
+// that with the annotation if provided.
+func (h *Handler) prometheusScrapePath(pod *corev1.Pod) (string, error) {
+	defaultPrometheusScrapePath := h.DefaultPrometheusScrapePath
+
+	if raw, ok := pod.Annotations[annotationPrometheusScrapePath]; ok && raw != "" {
+		return raw, nil
+	}
+
+	return defaultPrometheusScrapePath, nil
```

Comment:
```suggestion
	if raw, ok := pod.Annotations[annotationPrometheusScrapePath]; ok && raw != "" {
		return raw, nil
	}
	return h.DefaultPrometheusScrapePath, nil
```

Reply:
Also, do we need to return an error here? It doesnt look like it ever gets set.

Reply:
Nope we don't need an error for that method, I'll remove it